model_api/python/openvino/model_api/models/classification.py

```diff
@@ -243,9 +243,13 @@ def get_multiclass_predictions(self, outputs):
 def addOrFindSoftmaxAndTopkOutputs(inference_adapter, topk, output_raw_scores):
     nodes = inference_adapter.model.get_ops()
     softmaxNode = None
+    all_output_names = []
+    for output in inference_adapter.model.outputs:
+        all_output_names.extend(output.names)
     for op in nodes:
         if "Softmax" == op.get_type_name():
-            softmaxNode = op
+            if op.get_friendly_name() in all_output_names:
+                softmaxNode = op
```

Comment:
`ov::Model` has `const ov::ResultVector& get_results()` where `using ResultVector = std::vector<std::shared_ptr<ov::op::v0::Result>>;` and `Result` is a subclass of `Op` which is a subclass of `Node`.

Can you try replacing `nodes = inference_adapter.model.get_ops()` with `inference_adapter.model.get_results()`

Reply:
Well, I guess we can consider there is no other way. So I'm ok with this approach. Just make `all_output_names` to be a set, not list

Reply:
I implemented another version of workaround, @Wovchena please have a look